client/boot/index.js

```diff
@@ -383,15 +383,13 @@ function reduxStoreReady( reduxStore ) {
 	 * Layouts with differing React mount-points will not reconcile correctly,
 	 * so remove an existing single-tree layout by re-rendering if necessary.
 	 *
-	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 * TODO (@seear): Calypso-wide single tree rendering make this unnecessary
 	 */
 	page( '*', function( context, next ) {
-		const sectionNotIsomorphic = ! isSectionIsomorphic( context.store.getState() );
-		const previousLayoutIsSingleTree = ! isEmpty(
-			document.getElementsByClassName( 'wp-singletree-layout' )
-		);
+		const singleTreeSections = [ 'theme', 'themes' ];
+		const sectionName = getSectionName( context.store.getState() );
 
-		if ( sectionNotIsomorphic && previousLayoutIsSingleTree ) {
+		if ( includes( singleTreeSections, sectionName ) ) {
```

Comment:
This is calling re-render in the opposite direction to how it was called before. You can see that form the debug line below, which is now wrong. It does fix #6990, but leaves us with root replacement warnings.

Compare with branch fix/6990, which also fixes #6990, but without the warnings.